cpp/CMakeLists.txt

```diff
@@ -145,11 +135,52 @@ rapids_export(
   DOCUMENTATION doc_string
 )
 
+set(build_code_string
+    [=[
+
+# Find cuFile and determine which features are supported
+find_package(cuFile)
+if(NOT cuFile_FOUND)
+  message(WARNING "KvikIO: Building KvikIO without cuFile")
+else()
+  file(READ "${cuFile_INCLUDE_DIRS}/cufile.h" CUFILE_H_STR)
+  string(FIND "${CUFILE_H_STR}" "cuFileBatchIOSetUp" cuFileBatchIOSetUp_location)
+  if(cuFileBatchIOSetUp_location EQUAL "-1")
+    set(cuFile_BATCH_API_FOUND FALSE)
+  else()
+    set(cuFile_BATCH_API_FOUND TRUE)
+  endif()
+  message(STATUS "KvikIO: Found cuFile's Batch API: ${cuFile_BATCH_API_FOUND}")
+  string(FIND "${CUFILE_H_STR}" "cuFileReadAsync" cuFileReadAsync_location)
+  if(cuFileReadAsync_location EQUAL "-1")
+    set(cuFile_STREAM_API_FOUND FALSE)
+  else()
+    set(cuFile_STREAM_API_FOUND TRUE)
+  endif()
+  message(STATUS "KvikIO: Found cuFile's Stream API: ${cuFile_STREAM_API_FOUND}")
+endif()
+
+# Enable supported cuFile features in KvikIO
+if(cuFile_FOUND)
```

Comment:
You are going to keep adding these defines, includes, link libraries each time `find_package` is called. We want to do this as rarely as possible,

One option is to wrap the whole check in something along the lines of:
```cmake
get_property(already_set_kvikio DIRECTORY PROPERTY kvikio_already_set_defines SET)
if(NOT already_set_kvikio)
  set_property(DIRECTORY PROPERTY kvikio_already_set_defines "ON")
   find_package(cuFile)
   ...
endif()
```